metrics/metrics/src/main/java/io/helidon/metrics/Registry.java

```diff
@@ -225,12 +236,14 @@ public SimpleTimer simpleTimer(Metadata metadata, Tag... tags) {
      */
     @Override
     public synchronized boolean remove(String name) {
-        final List<MetricID> metricIDs = allMetricIDsByName.get(name);
-        if (metricIDs == null) {
+        final List<Map.Entry<MetricID, HelidonMetric>> doomedMetrics = getMetricsByName(name);
```

Comment:
Just a note: this seems like a large block of work to use `synchronized` on. I haven't been following all the locking semantics in here but maybe this could be chunked up a little more? Would that help with performance, or is this not in the hot path?

Reply:
The good news is that `remove` happens very rarely, certainly not during typical workloads.